src/pages/api/iexec/fund-release.tsx

```diff
@@ -0,0 +1,121 @@
+import mongoose from 'mongoose';
+import { getNewPayment } from '../../../queries/payments';
+import { EmailType, IPayment, PaymentTypeEnum, Web3mailPreferences } from '../../../types';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { sendMailToAddresses } from '../../../scripts/iexec/sendMailToAddresses';
+import { getUserWeb3mailPreferences } from '../../../queries/users';
+import { calculateCronData } from '../../../modules/Web3mail/utils/cron-utils';
+import {
+  checkPaymentExistenceInDb,
+  persistCronProbe,
+  persistEmail,
+} from '../../../modules/Web3mail/utils/database-utils';
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const chainId = process.env.NEXT_PUBLIC_NETWORK_ID;
+  const mongoUri = process.env.NEXT_MONGO_URI;
+  const cronSecurityKey = req.query.key;
+  const RETRY_FACTOR = 5;
+  let successCount = 0,
+    errorCount = 0;
+
+  if (cronSecurityKey !== process.env.NEXT_PRIVATE_CRON_KEY) {
+    return res.status(401).json({ message: 'Unauthorized' });
+  }
+
+  if (!chainId) {
+    return res.status(500).json('Chain Id is not set');
+  }
+
+  if (!mongoUri) {
+    return res.status(500).json('MongoDb URI is not set');
+  }
+
+  await mongoose.connect(mongoUri as string);
+
+  const platformId = process.env.NEXT_PUBLIC_PLATFORM_ID;
+
+  if (!platformId) {
+    throw new Error('Platform Id is not set');
+  }
+
+  // Check whether the user provided a timestamp or if it will come from the cron config
+  const { sinceTimestamp, cronDuration } = calculateCronData(
+    req,
+    RETRY_FACTOR,
+    EmailType.FundRelease,
+  );
+  try {
+    const response = await getNewPayment(Number(chainId), platformId, sinceTimestamp);
+    const nonSentPaymentEmails: IPayment[] = [];
+
+    // Check if some entities are not already in the DB
+    if (response.data.data.payments.length > 0) {
+      for (const payment of response.data.data.payments as IPayment[]) {
+        await checkPaymentExistenceInDb(payment, nonSentPaymentEmails, EmailType.FundRelease);
+      }
+    }
+
+    // If some entities are not already in the DB, email the hirer & persist the payment in the DB
+    if (nonSentPaymentEmails) {
+      for (const payment of nonSentPaymentEmails) {
+        let handle = '',
+          action = '',
+          address = '';
+        payment.paymentType === PaymentTypeEnum.Release
```

Comment:
ternair is not optimal for this use case, code would be more clear and clean with a proper if/elseIf,